From: Piotr Luc <piotr.luc@intel.com>
Date: Wed, 12 Oct 2016 20:27:58 +0200
Subject: perf/x86/intel/uncore: Add Knights Mill CPUID
Git-commit: ba2f81575eba8dcf128354169c20ae23f810f652
Patch-mainline: v4.9-rc2
References: FATE#321922
Signed-off-by: Tony Jones <tonyj@suse.de>

    [suse:
     MAJOR adjustment for context; no e633c65a Make the Intel uncore PMU driver
     modular

     adjust for context, no ef5f9f47d4ec4cf42bac48c7c4dafacc1b9f0630
       perf/x86/intel: Use Intel family macros for core perf events

     adjust for missing 6bcb2db547be8263a98ae9413127df9385b38763
     perf/x86: Move perf_event_intel_uncore.[ch] .. => 
        x86/events/intel/uncore.[ch]
    ]

    perf/x86/intel/uncore: Add Knights Mill CPUID
    
    Add Knights Mill (KNM) to the list of CPUIDs supported by PMU.
    
    Signed-off-by: Piotr Luc <piotr.luc@intel.com>
    Reviewed-by: Dave Hansen <dave.hansen@intel.com>
    Cc: Andy Lutomirski <luto@kernel.org>
    Cc: Borislav Petkov <bp@alien8.de>
    Cc: Brian Gerst <brgerst@gmail.com>
    Cc: Denys Vlasenko <dvlasenk@redhat.com>
    Cc: H. Peter Anvin <hpa@zytor.com>
    Cc: Josh Poimboeuf <jpoimboe@redhat.com>
    Cc: Linus Torvalds <torvalds@linux-foundation.org>
    Cc: Peter Zijlstra <peterz@infradead.org>
    Cc: Thomas Gleixner <tglx@linutronix.de>
    Link: http://lkml.kernel.org/r/20161012182758.2925-1-piotr.luc@intel.com
    Signed-off-by: Ingo Molnar <mingo@kernel.org>

---
 arch/x86/kernel/cpu/perf_event_intel_uncore.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/x86/kernel/cpu/perf_event_intel_uncore.c
+++ b/arch/x86/kernel/cpu/perf_event_intel_uncore.c
@@ -1011,6 +1011,7 @@ static int __init uncore_pci_init(void)
 		ret = bdw_uncore_pci_init();
 		break;
 	case 87: /* Knights Landing */
+        case 133: /* Knights Mill -- INTEL_FAM6_XEON_PHI_KNM */
 		ret = knl_uncore_pci_init();
 		break;
 	case 78: /* Skylake mobile -- INTEL_FAM6_SKYLAKE_MOBILE */
@@ -1319,6 +1320,7 @@ static int __init uncore_cpu_init(void)
 		bdx_uncore_cpu_init();
 		break;
 	case 87: /* Knights Landing */
+	case 133: /* Knights Mill -- INTEL_FAM6_XEON_PHI_KNM */
 		knl_uncore_cpu_init();
 		break;
         case 78: /* Skylake mobile -- INTEL_FAM6_SKYLAKE_MOBILE */
